### PR TITLE
 [MM-14283] Added E2E test for compact message display

### PR DESCRIPTION
**Pull request by thekiiingbob**
#### Summary
- Added test for compact message display setting
- Added new user to the users.json (This feels like a bandaid, as I think as the test suite grows there will be more occurrences of users stepping on each others toes)
- Improved/fixed the `toAccountSettingsModal()` function to default to user-1, but accept an alternate user as well.

#### Ticket Link
[GitHub Issue](https://github.com/mattermost/mattermost-server/issues/10322)
[JIRA Ticket](https://mattermost.atlassian.net/browse/MM-14283)
[Please link the GitHub issue or Jira ticket this PR addresses.]

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: cypress/integration/account_settings/display/message_display_mode_spec.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        cy.get('#message_displayFormatB').click();
+
+        // 7. Click "Save"
+        cy.get('#saveSetting').click();
+
+        // 8. Click "x" button to close Account Settings modal
+        cy.get('#accountSettingsHeader > .close').click();
+
+        // 9. Go to channel which has any posts
+        cy.visit('/ad-1/channels/ratione-1');
```
Comment (saturninoabril) [inline cypress/integration/account_settings/display/message_display_mode_spec.js]: I think it's possible to remove this step.

File: cypress/integration/account_settings/display/message_display_mode_spec.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        // 10. Enter in text
+        cy.get('#post_textbox').type('First line');
+        cy.get('#post_textbox').type('{shift}{enter}{enter}');
+        cy.get('#post_textbox').type('Text after');
+
+        // 11. Submit post
+        cy.get('#create_post').submit();
```
Comment (saturninoabril) [inline cypress/integration/account_settings/display/message_display_mode_spec.js]: Please combine steps 10/11 into:
```javascript
cy.get('#post_textbox').type('First line').type('{shift}{enter}{enter}').type('Text after{enter}');
```

File: cypress/integration/account_settings/display/message_display_mode_spec.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            cy.get('#edit_textbox').type(',edited');
+
+            // 16. Save
+            cy.get('#editButton').click();
+
+            // * Verify HTML includes newline and the edit
+            cy.get(postMessageTextId).should('have.html', '<p>First line</p>\n<p>Text after,edited</p>\n');
+
+            // *  Post should have (edited)
+            cy.get(`#post_${postId}`).find('.post-edited__indicator');
```
Comment (saturninoabril) [inline cypress/integration/account_settings/display/message_display_mode_spec.js]: Please add ID to [edited indicator](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_message_view/post_message_view.jsx#L119), preferably ``id={`postEdited_${this.props.post.id}`}`` so we can check it as:
```javascript
 cy.get(`#postEdited_${postId}`).should('be.visible').should('contain', '(edited)');
```

File: cypress/integration/account_settings/display/message_display_mode_spec.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        });
+    });
```
Comment (saturninoabril) [inline cypress/integration/account_settings/display/message_display_mode_spec.js]: As I mentioned above, please add step that reverts change on account setting.

File: cypress/fixtures/users.json [hunk 5]
```diff
@@ -12,5 +12,12 @@
         "email": "user-1@sample.mattermost.com",
         "firstName": "Victor",
         "lastName": "Welch"
+    },
+    "user-2": {
+        "username": "samuel.tucker",
+        "password": "user-2",
+        "email": "user-2@sample.mattermost.com",
+        "firstName": "Samuel",
+        "lastName": "Tucker"
```
[the remark this answers is not in the document]
Comment (thekiiingbob) [reply cypress/fixtures/users.json]: The change to the account setting for user-1 was impacting `account_settings/display/channel_display_mode_spec.js`.

In another comment you mention reverting the setting, which we can do, but if this test fails in a manner that makes it so we can't revert the setting then tests that run after this one may fail due to the setting not getting reverted.

I've usually handled this by either using distinct users for each test (or creating users for each test, during execution). Or setting the account settings to a default state in a before/beforeEach block in other tests (hopefully via API or something, so that we can avoid having to drive the UI each time). 

Let me know what you want for now and I can get it done either way.

[the remark this answers is not in the document]
Comment (saturninoabril) [reply cypress/fixtures/users.json]: Not sure how the change on `channel_display_mode_spec.js` is affecting this test but definitely, the change on that spec should be reverted as well.  Nice catch!

I totally agree on "test failed and setting not getting reverted" but we might end up on "stop on first fail" so that test will not continue on fail.

Use of distinct users is an easy pick but I prefer the reset of account settings via API.  I have a newly created ticket to address that - [MM-14391](https://mattermost.atlassian.net/browse/MM-14391).  I also have another ticket to add command in changing specific account setting - [MM-14390](https://mattermost.atlassian.net/browse/MM-14390).

At the moment, let's stick with "user-1" with this flow - change account setting via UI -> do the test on subject (e.g. post message) -> revert the change on account setting via UI.  My goal is to observe and address the Cypress limitation in driving the UI on this early stage of UI automation, and understand what it takes to be flaky.

[the remark this answers is not in the document]
Comment (thekiiingbob) [reply cypress/fixtures/users.json]: It had to do with how, when compact, the width of the post changes and that test happened to be checking specific sizes.

All changes should be pushed up now, including the latest ESLint fix.

File: cypress/support/commands.js [hunk 6]
```diff
@@ -257,8 +300,12 @@ Cypress.Commands.add('closeRHS', () => {
 
 Cypress.Commands.add('createNewTeam', (teamName, teamURL) => {
     cy.visit('/create_team');
-    cy.get('#teamNameInput').type(teamName).type('{enter}');
```
Comment (sudheerDev) [inline cypress/support/commands.js]: @thekiiingbob Nit: Can you revert the changes from single line to multi line commands?. They are fine as they were and it would be nice to have `git blame` point to the commit which made functional changes.

Comment (thekiiingbob) [reply cypress/support/commands.js]: @sudheerDev Should be all reverted now!

Comment (sudheerDev) [reply cypress/support/commands.js]: @thekiiingbob Thanks for reverting just one more thing. If you can merge master and update the PR i can merge this.  Mention me or saturninoabril when you update so we don't miss merging this.

**Comment by thekiiingbob** [comment]
@sudheerDev @saturninoabril Master has been merged in.

**Comment by sudheerDev** [comment]
@thekiiingbob Merged 🎉
